### ISSUE TITLE
:find typing full file without tab completion changes case of file

### ISSUE THREAD

**Issue by jmzagorski** [vim/vim#6088]
**Describe the bug**
When searching for a file, for example Foobar.cs, using `:find path/to/foobar.cs<CR>`, the file casing gets loaded into the buffer in lower case and changes the casing of the original file on save. Experiencing this in vim/gvim.

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Run `vim --clean` (or `gvim --clean`, etc.)
2. `:find path/to/foobar.cs<CR>`
3. If the file name is PascalCase, for example, it is changed to whatever casing you used. In my example lowsercase

**Expected behavior**
The casing should match the file name and not change it.

**Environment (please complete the following information):**
Windows 10 64 bit. Version compiled by me on windows.

![image](https://user-images.githubusercontent.com/8404957/82082560-3aa00000-96b6-11ea-8485-e39717ec3b34.png)

**Comment by ivandev-81** [comment]
Hi @jmzagorski , Is this still an open problem ? I just have reproduced that bug but it doesn't appear to me, instead Vim says that the file doesn't exist, Do you have used it in Linux or Windows ?

**Comment by jmzagorski** [comment]
Yes this is still an issue. I am on windows. Your error makes sense if you are on Linux since file names are case sensitive. Thanks for letting me know that I am not the only one!

**Comment by ivandev-81** [comment]
@jmzagorski Yes, it seems to be the problem: The subyacente file system, I'm using Linux is the reason for I can't reproduce this bug, (another point for Linux supporters hehe) but try use Vim (not GVim) from a "bash terminal emulator in Windows" like MinGW directly (no GVim) to check if is the file system indeed.
